packages/core/src/app/shipping/shippingOption/ShippingOptionsForm.tsx

```diff
@@ -30,16 +44,22 @@ class ShippingOptionsForm extends PureComponent<
         this.unsubscribe = subscribeToConsignments(this.selectDefaultShippingOptions);
     }
 
-    componentDidUpdate(): void {
+    componentDidUpdate({ shippingFormRenderTimestamp }: ShippingOptionsFormProps): void {
         const {
             analyticsTracker,
             consignments,
-            shouldShowShippingOptions
+            shouldShowShippingOptions,
+            shippingFormRenderTimestamp: newShippingFormRenderTimestamp,
+            setValues,
         } = this.props;
 
         if (consignments?.length && shouldShowShippingOptions) {
             analyticsTracker.showShippingMethods();
         }
+
+        if (newShippingFormRenderTimestamp !== shippingFormRenderTimestamp) {
+            setValues(getShippingOptionIds(this.props));
+        }
```

Comment:
I was unable to cover this with a unit test because I couldn't trigger `componentDidUpdate` in Enzyme.